src/pydvl/influence/general.py

```diff
@@ -39,88 +22,107 @@ class InfluenceType(str, Enum):
-    :param x_train: A np.ndarray of shape [MxK] containing the features of the input data points.
-    :param y_train: A np.ndarray of shape [MxL] containing the targets of the input data points.
-    :param x_test: A np.ndarray of shape [NxK] containing the features of the test set of data points.
-    :param y_test: A np.ndarray of shape [NxL] containing the targets of the test set of data points.
-    :param inversion_func: function to use to invert the product of hvp (hessian vector product) and the gradient
-        of the loss (s_test in the paper).
+    It calculates influence factors of a model from its training and test
+    points. Given a test point $z_test = (x_{test}, y_{test})$, a loss
+    $L(z_{test}, \theta)$ ($\theta$ being the parameters of the model) and the
+    Hessian of the model $H_{\theta}$, influence factors are defined as
+    $s_{test} = H_{\theta}^{-1} \grad_{\theta} L(z_{test}, \theta)$. They are
+    used for efficient influence calculation. This method first
+    (implicitly) calculates the Hessian and then (explicitly) finds the
+    influence factors for the model using the given inversion method. The
+    parameter ``lam`` is used to regularize the inversion of the Hessian. For
+    more info, refer to :footcite:t:`koh_understanding_2017`, paragraph 3.
+
+    :param model: A model wrapped in the TwiceDifferentiable interface.
+    :param x_train: An array of shape [MxK] containing the features of the input data points.
+    :param y_train: An array of shape [MxL] containing the targets of the input data points.
+    :param x_test: An array of shape [NxK] containing the features of the test set of data points.
+    :param y_test: An array of shape [NxL] containing the targets of the test set of data points.
+    :param inversion_func: function to use to invert the product of hvp (hessian
+        vector product) and the gradient of the loss (s_test in the paper).
     :param lam: regularization of the hessian
     :param progress: If True, display progress bars.
-    :returns: A np.ndarray of size (N, D) containing the influence factors for each dimension (D) and test sample (N).
+    :returns: An array of size (N, D) containing the influence factors for each
+        dimension (D) and test sample (N).
     """
-    if not _TORCH_INSTALLED:
-        raise RuntimeWarning("This function requires PyTorch.")
     grad_xy, _ = model.grad(x, y)
     hvp = lambda v: model.mvp(grad_xy, v) + lam * v
+    n_params = model.num_params()
     test_grads = model.split_grad(x_test, y_test, progress)
-    return inversion_func(hvp, test_grads)
+    return invert_matrix(
+        inversion_method,
+        hvp,
+        (n_params, n_params),
+        test_grads,
+        progress,
+    )
 
 
 def _calculate_influences_up(
     model: TwiceDifferentiable,
-    x: "NDArray",
-    y: "NDArray",
-    influence_factors: "NDArray",
+    x: TensorType,
+    y: TensorType,
+    influence_factors: NDArray,
```

Comment:
Why not a TensorType as the rest? It would make more sense IMO to remain within the framework used.